sled-agent/src/bootstrap/agent.rs

```diff
@@ -481,10 +493,38 @@ impl Agent {
     }
 }
 
+// We intentionally DO NOT derive `Debug` or `Serialize`; both provide avenues
+// by which we may accidentally log the contents of our trust quorum share.
+#[derive(Deserialize, PartialEq)]
+struct PersistentSledAgentRequest<'a> {
+    request: Cow<'a, SledAgentRequest>,
+    trust_quorum_share: Option<SerializableShareDistribution>,
+}
+
+impl PersistentSledAgentRequest<'_> {
+    /// On success, the returned string will contain our raw
+    /// `trust_quorum_share`. This method is named `danger_*` to remind the
+    /// caller that they must not log this string.
+    fn danger_serialize_as_toml(&self) -> Result<String, toml::ser::Error> {
+        #[derive(Serialize)]
+        #[serde(remote = "PersistentSledAgentRequest")]
+        struct PersistentSledAgentRequestDef<'a> {
```

Comment:
What does the `Def` suffix stand for?

Reply:
"Definition", I think; this is from https://serde.rs/remote-derive.html. In this case we don't _need_ remote derive because we control the original type, but we choose to use it because the original type intentionally doesn't derive `Serialize`.

Reply:
Ah, cool it comes from serde example. Stick with it then!